core/embed/rust/src/ui/display/font.rs

```diff
@@ -198,31 +202,28 @@ impl Font {
             return 0;
         }
 
-        if let Some(glyph) = self.get_first_glyph_from_text(text) {
-            glyph.bearing_x
-        } else {
-            0
-        }
+        self.get_first_glyph_from_text(text)
+            .map_or(0, |glyph| glyph.bearing_x)
     }
 
     pub fn char_width(self, ch: char) -> i16 {
-        display::char_width(ch, self.into())
+        let mut buf = [0u8; 4];
+        let encoding = ch.encode_utf8(&mut buf);
+        self.text_width(encoding)
```

Comment:
you should be able to implement this directly as `self.get_glyph(c).adv`, and compute `text_width` by mapping to `char_width`
(btw is `char_width` used anywhere?)

Reply:
4660ea4d48725ac94a70b862201cc815abb63d10